pull.go

```diff
@@ -292,7 +292,9 @@ func pullImage(ctx context.Context, store storage.Store, srcRef types.ImageRefer
 	}()
 
 	logrus.Debugf("copying %q to %q", transports.ImageName(srcRef), destName)
-	if _, err := retryCopyImage(ctx, policyContext, maybeCachedDestRef, srcRef, srcRef, getCopyOptions(store, options.ReportWriter, sc, nil, "", options.RemoveSignatures, "", nil, nil, options.OciDecryptConfig), options.MaxRetries, options.RetryDelay); err != nil {
+	destinationCtx := getSystemContext(store, nil, "")
+	destinationCtx.BigFilesTemporaryDir = sc.BigFilesTemporaryDir
```

Comment:
Why not set `BigFilesTemporaryDir` in `getSystemContext()`?  Why don't we reuse the source context or use it as the source of defaults when creating `destinationCtx`?

Reply:
FWIW there’s a bit of history to this, it turns out: The original was #2022, accidentally broken in #2868 (but the pull case that is the focus of this PR was probably never handled correctly).

---

I agree that the flow of defaults/options is counterintuitive and likely wrong; just like the value of `DockerRegistryUserAgent` set in `SystemContextFromOptions`, this is supposed to be a codebase-wide default that applies to ~everything, not something only related to processing of CLI options.

Having defaults in `getSystemContext` would make more sense for the core c/buildah code, OTOH not all users of a `*types.SystemContext` do pass through `getSystemContext` (e.g. not the uses in `c/cmd/buildah/common.go`) — that might not matter if all of them are just reads of c/storage metadata.

---

One concern with using the same `SystemContext` for both the source and destination of a copy is that `SystemContextFromOptions` may set `DockerAuthConfig`, which is not explicitly associated with a registry, and could cause credentials intended for the source to be used in the destination, or vice versa. I don’t expect that can currently happen (can Buildah cause a `docker://`-`docker://` copy?), but it’s a risk for the general idea.

Reply:
Ok I simplified it to just push in the source as the destination though the  getSystemContext(store, sc, "")